packages/salesforcedx-vscode-apex-replay-debugger/test/unit/activation/getDialogStartingPath.test.ts

```diff
@@ -0,0 +1,106 @@
+import {
+  projectPaths,
+  workspaceUtils
+} from '@salesforce/salesforcedx-utils-vscode';
+import * as pathExists from 'path-exists';
+import * as vscode from 'vscode';
+import { getDialogStartingPath } from '../../../src/activation/getDialogStartingPath';
+
+describe('getDialogStartingPath', () => {
+  const testPath = '/here/is/a/fake/path/to/';
+  let hasRootWorkspaceStub: jest.SpyInstance;
+  let pathExistsMock: jest.SpyInstance;
+  let vsCodeUriMock: jest.SpyInstance;
+  let debugLogsFolderMock: jest.SpyInstance;
+  let stateFolderMock: jest.SpyInstance;
+
+  beforeEach(() => {
+    hasRootWorkspaceStub = jest.spyOn(workspaceUtils, 'hasRootWorkspace');
+    pathExistsMock = jest.spyOn(pathExists, 'sync').mockReturnValue(true);
+    vsCodeUriMock = jest.spyOn(vscode.Uri, 'file');
+    debugLogsFolderMock = jest.spyOn(projectPaths, 'debugLogsFolder');
+    stateFolderMock = jest.spyOn(projectPaths, 'stateFolder');
+  });
+
+  it('Should return last opened log folder if present', () => {
+    hasRootWorkspaceStub.mockReturnValue(true);
+    const mockGet = jest.fn().mockReturnValue(testPath);
+    const mockExtensionContext: any = { workspaceState: { get: mockGet } };
+    vsCodeUriMock.mockReturnValue({ path: testPath } as vscode.Uri);
+
+    // Act
+    const dialogStartingPathUri = getDialogStartingPath(mockExtensionContext);
+
+    expect(hasRootWorkspaceStub).toHaveBeenCalled();
+    expect(mockGet).toHaveBeenCalledWith(LAST_OPENED_LOG_FOLDER_KEY);
+    expect(pathExistsMock).toHaveBeenCalledWith(testPath);
+    expect(vsCodeUriMock).toHaveBeenCalledWith(testPath);
+    expect((dialogStartingPathUri as vscode.Uri).path).toEqual(testPath);
+  });
+
+  it('Should return project log folder when last opened log folder not present', async () => {
+    hasRootWorkspaceStub.mockReturnValue(true);
+    const mockGet = jest.fn().mockReturnValue(undefined);
+    const mockExtensionContext: any = { workspaceState: { get: mockGet } };
+    const fakePathToDebugLogsFolder = 'path/to/debug/logs';
+    debugLogsFolderMock.mockReturnValue(fakePathToDebugLogsFolder);
+    vsCodeUriMock.mockReturnValue({
+      path: fakePathToDebugLogsFolder
+    } as vscode.Uri);
+
+    // Act
+    const dialogStartingPathUri = getDialogStartingPath(mockExtensionContext);
+
+    expect(hasRootWorkspaceStub).toHaveBeenCalled();
+    expect(mockGet).toHaveBeenCalledWith(LAST_OPENED_LOG_FOLDER_KEY);
+    expect(pathExistsMock).toHaveBeenCalledWith(fakePathToDebugLogsFolder);
+    expect(vsCodeUriMock).toHaveBeenCalledWith(fakePathToDebugLogsFolder);
+    expect((dialogStartingPathUri as vscode.Uri).path).toEqual(
+      fakePathToDebugLogsFolder
+    );
+  });
+
+  it('Should return state folder as fallback when project log folder not present', async () => {
+    hasRootWorkspaceStub.mockReturnValue(true);
+    const mockGet = jest.fn().mockReturnValue(undefined);
+    const mockExtensionContext: any = { workspaceState: { get: mockGet } };
```

Comment:
total nit: but any time something exists in all tests it should probably be pulled up to the beforeEach.